domain/src/Plugin/Condition/Domain.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\domain\Plugin\Condition\Domain.
+ */
+
+namespace Drupal\domain\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\Context\ContextDefinition;
+
+/**
+ * Provides a 'Domain' condition.
+ *
+ * @Condition(
+ *   id = "domain",
+ *   label = @Translation("Domain")
+ * )
+ *
+ */
+class Domain extends ConditionPluginBase {
```

Comment:
Is this really the core pattern? To name the condition after the module and not DomainCondition? That seems odd.